lib/report.js

```diff
@@ -158,7 +160,155 @@ class ConciseReporter extends Reporter {
+    // if we should do this, i.e. store map of successful tests
+    // this.successes--;
+    // test.todo ? this.skipped++ : this.failures++;
+    // TODO(lundibundi): add test.metadata.filename here too once available
+    this.listFailure(
+      test,
+      { message: `Failure after end: ${error ? error.message : 'no message'}` },
+      ` Test failed: ${test.caption}`
+    );
+  }
+
+  lastElement(inc) {
+    if (inc) this.testNumeration[this.lastIndex]++;
+    return this.testNumeration[this.lastIndex];
+  }
+
+  parseTestResults(test) {
+    this.lastElement(1);
+    if (!test.success) {
+      if (!test.todo) {
+        this.failedTests.push(this.testNumeration[this.lastIndex] - 1);
+      }
+      this.listFailure(test, true);
+      this.got.forEach(el => {
+        this.message += `${pad(6)}- ${el}\n`;
+      });
+      this.message += `${pad(4)}expect: \n`;
+      this.expected.forEach(el => {
+        this.message += `${pad(6)}- ${el}\n`;
+      });
+      this.message += '  ...\n';
+      this.got = [];
+      this.expected = [];
+      console.log(this.message);
+    } else {
+      this.testNumeration.push(0);
+      this.lastIndex++;
+      this.message = pad(this.lastIndex * 2);
+      test.results.forEach(res => {
+        if (res.success) {
+          if (res.type === 'subtest') {
+            const t = res.test;
+            console.log(`${this.message}# Subtest: ${t.caption}`);
+            this.parseTestResults(t);
+            this.message = pad(this.lastIndex * 2);
+            const prefix = t.success ? 'ok' : 'not ok';
+            let message = `${this.message}${prefix} ${this.lastElement(0)}`;
+            message += ` - ${t.caption}`;
+            console.log(message);
+          } else {
+            this.listSuccess(res);
+          }
+        }
+      });
+      console.log(`${this.message}1..${this.lastElement(0)}`);
+      this.lastIndex--;
+      this.testNumeration.pop();
+    }
+    if (!this.lastIndex) this.message = '\n';
+  }
+
+  listSuccess(test) {
+    const testPrefix = test.success ? 'ok' : 'not ok';
+    console.log(`${this.message}${testPrefix} ${this.lastElement(1)}`);
+  }
+
+  listFailure(test, firstFailure) {
+    test.results.forEach(res => {
+      if (firstFailure) {
+        firstFailure = false;
+        this.message += '  ---\n';
+        if (res.message) this.message += `  message: '${res.message}'\n`;
+        this.message += '  severity: ';
+        this.message += res.severity ? `${res.severity}\n` : 'fail\n';
+        this.message += `${pad(2)}data:\n`;
+        this.message += `${pad(4)}got:\n`;
+      }
+      const [expected, actual] = [res.expected, res.actual].map(res => (
+        typeof res === 'function' || res instanceof Error ?
+          res.toString() :
+          res
+      ));
+      if (expected === undefined && actual === undefined) {
+        this.listFailure(res.test);
+      } else {
+        this.got.push(expected);
```

Comment:
Expected and resulting values should be pushed to different arrays. This is obvious even from variables names.
```suggestion
        this.got.push(actual);
         this.expected.push(expected);
```

Reply:
I don't know what the matter of this case, but it works correct. Try it yourself!

Reply:
No, it is not. Values are reverted in the output.